prettyprinter/src/Data/Text/Prettyprint/Doc/Internal.hs

```diff
@@ -1785,7 +1794,16 @@ layoutWadlerLeijen
       | fits pWidth minNestingLevel availableWidth x = x
       | otherwise = y
       where
-        minNestingLevel = min lineIndent currentColumn
+        minNestingLevel =
+            -- See https://github.com/quchen/prettyprinter/issues/83.
+            if startsWithLine y
+                -- y might be a (more compact) hanging layout. Let's check x
+                -- thoroughly with the smaller lineIndent.
```

Comment:
If we wanted to detect hanging layouts more precisely, we could also check that the indentation of the `SLine` is `< currentColumn`. But what we have right now is the slightly more conservative change.